src/corehost/src/tpafile.cpp

```diff
@@ -57,195 +75,188 @@ bool read_field(pal::string_t line, int& offset, pal::string_t& value_recv)
     return true;
 }
 
-bool tpafile::load(pal::string_t path)
+bool tpafile::load(const pal::string_t& path)
 {
-    // Check if the file exists, if not, there is nothing to add
     if (!pal::file_exists(path))
     {
         return true;
     }
 
-    // Open the file
     pal::ifstream_t file(path);
     if (!file.good())
     {
-        // Failed to open the file!
         return false;
     }
 
-    // Read lines from the file
-    while (true)
+    std::string line;
+    while (std::getline(file, line))
     {
-        std::string line;
-        std::getline(file, line);
-        auto line_palstr = pal::to_palstring(line);
-        if (file.eof())
-        {
-            break;
-        }
-
-        auto offset = 0;
-
         tpaentry_t entry;
+        pal::string_t* fields[] = {
+            &entry.library_type,
+            &entry.library_name,
+            &entry.library_version,
+            &entry.library_hash,
+            &entry.asset_type,
+            &entry.asset_name,
+            &entry.relative_path
+        };
 
-        // Read fields
-        if (!(read_field(line_palstr, offset, entry.library_type))) return false;
-        if (!(read_field(line_palstr, offset, entry.library_name))) return false;
-        if (!(read_field(line_palstr, offset, entry.library_version))) return false;
-        if (!(read_field(line_palstr, offset, entry.library_hash))) return false;
-        if (!(read_field(line_palstr, offset, entry.asset_type))) return false;
-        if (!(read_field(line_palstr, offset, entry.asset_name))) return false;
-        if (!(read_field(line_palstr, offset, entry.relative_path))) return false;
+        auto line_palstr = pal::to_palstring(line);
+        std::vector<pal::char_t> buf(line_palstr.length());
 
-        m_entries.push_back(entry);
+        for (unsigned i = 0, offset = 0; i < sizeof(fields) / sizeof(fields[0]); ++i)
+        {
+            if (!(read_field(line_palstr, buf.data(), &offset, fields[i])))
+            {
+                return false;
+            }
+        }
+        m_tpa_entries.push_back(entry);
     }
-
     return true;
 }
 
-void tpafile::add_from_local_dir(const pal::string_t& dir)
+void tpafile::get_local_assemblies(const pal::string_t& dir)
```

Comment:
We should call this "PopulateTPAFromPath" since it is a general purpose function.